met-api/src/met_api/models/survey.py

```diff
@@ -32,45 +37,29 @@ class Survey(db.Model):  # pylint: disable=too-few-public-methods
     submissions = db.relationship('Submission', backref='survey', cascade='all, delete')
 
     @classmethod
-    def get_survey(cls, survey_id) -> SurveySchema:
+    def get_survey(cls, survey_id) -> Survey:
         """Get a survey."""
-        survey_schema = SurveySchema()
-        survey = db.session.query(Survey)\
```

Comment:
do we not need the engagement status data when getting a survey?

Reply:
We are accessing survye.engagement.status_id in survey building page for example:

    const hasEngagement = Boolean(savedSurvey?.engagement);
    const isEngagementDraft = savedSurvey?.engagement?.status_id === EngagementStatus.Draft;
    const hasPublishedEngagement = hasEngagement && !isEngagementDraft;

Reply:
My understanding is that the engagement is being loaded by a separate call in the service class

Reply:
const loadSurvey = async () => {
        if (isNaN(Number(surveyId))) {
            navigate('/survey/listing');
            dispatch(
                openNotification({
                    severity: 'error',
                    text: 'The survey id passed was erroneous',
                }),
            );
            return;
        }
        try {
            const loadedSurvey = await getSurvey(Number(surveyId));
            setSavedSurvey(loadedSurvey);
            setName(loadedSurvey.name);
            setLoading(false);
        } catch (error) {
            dispatch(
                openNotification({
                    severity: 'error',
                    text: 'Error occurred while loading saved survey',
                }),
            );
            navigate('/survey/listing');
        }
    };

Reply:
we're only calling getSurvey

Reply:
I am doing the join at later point of time in the service..
```    @classmethod
    def get(cls, survey_id):
        """Get survey by the id."""
        survey_model: SurveyModel = SurveyModel.get_survey(survey_id)
        engagement_model: EngagementModel = EngagementModel.get_engagement(survey_model.engagement_id)
        survey = SurveySchema().dump(survey_model)
        eng = EngagementSchema().dump(engagement_model)
        eng['banner_url'] = ObjectStorageService.get_url(engagement_model.banner_filename)
        survey['engagement'] = eng
        return survey

    @classmethod
    def get_open(cls, survey_id):
        """Get survey by the id."""
        survey_model = SurveyModel.get_open(survey_id)
        engagement_model: EngagementModel = EngagementModel.get_engagement(survey_model.engagement_id)
        survey = SurveySchema().dump(survey_model)
        eng = EngagementSchema().dump(engagement_model)
        eng['banner_url'] = ObjectStorageService.get_url(engagement_model.banner_filename)
        survey['engagement'] = eng
        return survey
```